### PR TITLE
Makefile

### PR DESCRIPTION
**Pull request by samzzzzzh**

### CODE REVIEW

**Review by Jaskaran1116 — COMMENTED** [review]
cleanedData should be cleaned_data

File: Makefile [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+
+
+all: notebooks/forest-fire-analysis.html
+
+# loading the data
+data/forestfires.csv : data/forestfires.csv src/R/data_cleaning.R
+		Rscript src/R/data_cleaning.R data/forestfires.csv data
+		
+# tidying the data
+data/cleanedData.csv : data/forestfires.csv src/R/data_tidying.R
```
Comment (a-kong) [inline Makefile:20]:
```suggestion
data/cleaned_data.csv : data/forestfires.csv src/R/data_tidying.R
```

File: Makefile [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+		Rscript src/R/data_tidying.R data/forestfires.csv data
+
+# data pre-processing: split into training and testing sets
+data/training_data.csv : data/cleanedData.csv src/R/data_splitting.R
+		Rscript src/R/data_splitting.R data/cleanedData.csv data
+
+data/testing_data.csv : data/cleanedData.csv src/R/data_splitting.R
+		Rscript src/R/data_splitting.R data/cleanedData.csv data
```
Comment (a-kong) [inline Makefile:28]:
```suggestion
		Rscript src/R/data_splitting.R data/cleaned_data.csv data
```

File: Makefile [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+# fit model - test model performance using RMSE metric
+results/testing_rmse.csv : data/training_data.csv data/testing_data.csv results/best_k.csv src/R/fit_model.R
+		Rscript src/R/fit_model.R data/training_data.csv data/testing_data.csv results/best_k.csv results
+
+# render report
+notebooks/forest-fire-analysis.html : notebooks/forest-fire-analysis.Rmd  results/best_k.csv results/testing_rmse.csv
+		Rscript -e "rmarkdown::render('notebooks/forest-fire-analysis.Rmd')"
+
+clean:
+		rm -f data/training_data.csv data/testing_data.csv
```
Comment (a-kong) [inline Makefile:50]: adding in the plot images
```suggestion
		rm -f data/training_data.csv data/testing_data.csv 
		rm -f results/correlation_graph.png results/correlation_graph_selected.png
```

**Review by Jaskaran1116 — APPROVED** [review]
otherwise, looks good to me ! great work !

File: Makefile [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+		Rscript src/R/fit_model.R data/training_data.csv data/testing_data.csv results/best_k.csv results
+
+# render report
+notebooks/forest-fire-analysis.html : notebooks/forest-fire-analysis.Rmd  results/best_k.csv results/testing_rmse.csv
+		Rscript -e "rmarkdown::render('notebooks/forest-fire-analysis.Rmd')"
+
+clean:
+		rm -f data/training_data.csv data/testing_data.csv 
+		rm -f results/correlation_graph.png results/correlation_graph_selected.png
+		rm -f data/scaled_and_centered_training_data.csv data/cleanedData.csv
```
Comment (a-kong) [inline Makefile:52]:
```suggestion
		rm -f data/scaled_and_centered_training_data.csv data/cleaned_data.csv
```

**Review by a-kong — APPROVED** [review]
one last thing!

**Pull request merged**